transitclock/src/main/java/org/transitclock/core/AvlProcessor.java

```diff
@@ -1583,19 +1584,19 @@ public void cacheAvlReportWithoutProcessing(AvlReport avlReport) {
 		VehicleState vehicleState =
 				VehicleStateManager.getInstance().getVehicleState(
 						avlReport.getVehicleId());
-		
+		ReentrantLock l = new ReentrantLock();
+		l.lock();
 		// Since modifying the VehicleState should synchronize in case another
 		// thread simultaneously processes data for the same vehicle. This
 		// would be extremely rare but need to be safe.
-		synchronized (vehicleState) {
-			// Update AVL report for cached VehicleState
-			vehicleState.setAvlReport(avlReport);
+		// Update AVL report for cached VehicleState
+		vehicleState.setAvlReport(avlReport);
 
-			// Let vehicle data cache know that the vehicle state was updated
-			// so that new IPC vehicle data will be created and cached and
-			// made available to the API.
-			VehicleDataCache.getInstance().updateVehicle(vehicleState);
-		}
+		// Let vehicle data cache know that the vehicle state was updated
+		// so that new IPC vehicle data will be created and cached and
+		// made available to the API.
+		VehicleDataCache.getInstance().updateVehicle(vehicleState);
+		l.unlock();

```

Comment:
again, if the updateVehicle(vehicleState) is at issue by being outside of the synchronized block, then consider moving it.